docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Regression/OnlineGradientDescent.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML.Data;
+
+namespace Microsoft.ML.Samples.Dynamic.Trainers.Regression
+{
+    public static class OnlineGradientDescent
+    {
+        public static void Example()
+        {
+            // Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
+            // as a catalog of available operations and as the source of randomness.
+            // Setting the seed to a fixed number in this example to make outputs deterministic.
+            var mlContext = new MLContext(seed: 0);
+
+            // Create a list of training data points.
+            var dataPoints = GenerateRandomDataPoints(1000);
+
+            // Convert the list of data points to an IDataView object, which is consumable by ML.NET API.
+            var trainingData = mlContext.Data.LoadFromEnumerable(dataPoints);
+
+            // Define the trainer.
+            var pipeline = mlContext.Regression.Trainers.OnlineGradientDescent();
+
+            // Train the model.
+            var model = pipeline.Fit(trainingData);
+
+            // Create testing data. Use different random seed to make it different from training data.
+            var testData = mlContext.Data.LoadFromEnumerable(GenerateRandomDataPoints(500, seed:123));
+
+            // Run the model on test data set.
+            var transformedTestData = model.Transform(testData);
+
+            // Convert IDataView object to a list.
+            var predictions = mlContext.Data.CreateEnumerable<Prediction>(transformedTestData, reuseRowObject: false).ToList();
+
+            // Look at 5 predictions
+            foreach (var p in predictions.Take(5))
+                Console.WriteLine($"Label: {p.Label:F3}, Prediction: {p.Score:F3}");
+
+            // Expected output:
+            //   OGD doesn't work: https://github.com/dotnet/machinelearning/issues/2425
```

Comment:
Wouldn't it be better to not put this on the official docs page?

Reply:
Also, the issue says that it doesn't have baselines, not that it doesn't work. Can you update the issue?

---
In reply to: [268338288](https://github.com/dotnet/machinelearning/pull/3067#discussion_r268338288) [](ancestors = 268338288)

Reply:
OGD works, there are 5 tests about it running regularly. 



---
In reply to: [268338288](https://github.com/dotnet/machinelearning/pull/3067#discussion_r268338288) [](ancestors = 268338288)

Reply:
How about we put the output in and add a `// TODO #ISSUE_NUMBER: Explanation` above the output.

---
In reply to: [268347907](https://github.com/dotnet/machinelearning/pull/3067#discussion_r268347907) [](ancestors = 268347907,268338288)